kubeadm/scripts/PrepareNode.ps1

```diff
@@ -51,9 +69,17 @@ DownloadFile $kubeletBinPath https://dl.k8s.io/$KubernetesVersion/bin/windows/am
 DownloadFile "$global:KubernetesPath\kubeadm.exe" https://dl.k8s.io/$KubernetesVersion/bin/windows/amd64/kubeadm.exe
 DownloadFile "$global:KubernetesPath\wins.exe" https://github.com/rancher/wins/releases/download/v0.0.4/wins.exe
 
-# Create host network to allow kubelet to schedule hostNetwork pods
-Write-Host "Creating Docker host network"
-docker network create -d nat host
+if ($ContainerRuntime -eq "Docker") {
+    # Create host network to allow kubelet to schedule hostNetwork pods
+    # NOTE: For containerd the 0-containerd-nat.json network config template added by
+    # Install-containerd.ps1 joins pods to the host network.
+    Write-Host "Creating Docker host network"
+    docker network create -d nat host
+} elseif ($ContainerRuntime -eq "containerD") {
+    DownloadFile "c:\k\hns.psm1" https://github.com/Microsoft/SDN/raw/master/Kubernetes/windows/hns.psm1
+    Import-Module "c:\k\hns.psm1"
+    New-HnsNetwork -Type NAT -Name nat
```

Comment:
When running through this I got the following error on this step: 
```
Invoke-HnsRequest : @{Error=A network with this name already exists. ; ErrorCode=2151350288; Success=False}                                             At C:\k\hns.psm1:233 char:16                                                                                                                            + ...      return Invoke-HnsRequest -Method POST -Type networks -Data $Json .
```

I think this maybe because containerd started with the nat network config during install containerd?

Reply:
This did not cause a fatal error just a warning that it was already created.

Reply:
I added a TODO in the code to come back and clean this up